rust/agents/relayer/src/msg/processor.rs

```diff
@@ -188,14 +140,22 @@ impl MessageProcessor {
                 "Sending message to submitter"
             );
             // Finally, build the submit arg and dispatch it to the submitter.
-            let submit_args = SubmitMessageArgs::new(message, checkpoint, proof);
-            self.tx_msg.send(submit_args)?;
+            let submit_args = SubmitMessageArgs::new(message.clone());
+            if let Some(send_channel) = self.send_channels.get(&message.destination) {
+                send_channel.send(submit_args)?;
+            } else {
+                warn!(
```

Comment:
May wanna lower this logging severity given that it's very possible relayers will not be wanting to process messages to all destinations

Reply:
Yes great point